archinstall/lib/menu/selection_menu.py

```diff
@@ -377,16 +250,44 @@ def _verify_selection_enabled(self, selection_name):
 				for d in selection.dependencies_not:
 					if not self._menu_options.get(d).is_empty():
 						return False
-
 			return True
 
 		raise ValueError(f'No selection found: {selection_name}')
 
-	def _menus_to_enable(self):
+	def _menus_to_enable(self) -> dict:
+		""" general """
 		enabled_menus = {}
 
 		for name, selection in self._menu_options.items():
 			if self._verify_selection_enabled(name):
 				enabled_menus[name] = selection
 
 		return enabled_menus
+
+	def option(self,name :str) -> Selector:
+		# TODO check inexistent name
+		return self._menu_options[name]
+
+	def set_option(self, name :str, selector :Selector):
+		self._menu_options[name] = selector
+
+	def _check_mandatory_status(self) -> bool:
+		for field in self._menu_options:
+			option = self._menu_options[field]
+			if option.is_mandatory() and not option.has_selection():
+				return False
+		return True
+
+	def set_mandatory(self, field :str, status :bool):
```

Comment:
> Instead of having this separate function, I'd propose to have an additional parameter in the `enable(...)` function with `enable(self, mandatory)`. With the additional function it is easy to forget the to set it, but with the mandatory parameter it has to be set.

The idea of using the *enable* method to add the status of mandatory, is a good one. But there is a reason why i want a set_mandatory method separated. This allows to define an option mandatory on the fly, depending of the needs of the application. 
OTOH when I do set_mandatory(True) I have to enable the field. I'try not to provoke a loop

Reply:
Done.  
I have changed the signature of Selector.set_enabled() to ```set_enabled(self, status :bool=True)``` to be able to switch on and off, and included an ```is_enabled()``` method, to get some measure of consistency